project/Build.scala

```diff
@@ -1008,9 +1040,76 @@ object Build {
         file.getPath.endsWith("scala-library-src/scala/Nothing.scala") ||
         file.getPath.endsWith("scala-library-src/scala/Null.scala") ||
         file.getPath.endsWith("scala-library-src/scala/Singleton.scala"))),
+      (Compile / sources) := {
+        val files = (Compile / sources).value
+        val overwritenSourcesDir = (Compile / scalaSource).value
+        val overwritenSources = files.flatMap(_.relativeTo(overwritenSourcesDir)).toSet
+        val reference = (Compile/sourceManaged).value / "scala-library-src"
+        files.filterNot(_.relativeTo(reference).exists(overwritenSources))
+      },
       (Test / managedClasspath) ~= {
         _.filterNot(file => file.data.getName == s"scala-library-${stdlibVersion(Bootstrapped)}.jar")
       },
+  )
+
+  /** Creates a copy of the Scala 2.13 library JAR with TASTy from `stdlib-2_13-bootstrapped` */
+  lazy val `stdlib-2_13-tasty-bundle` = project.in(file("stdlib-2_13-tasty-bundle")).
+    withCommonSettings(Bootstrapped).
+    dependsOn(`stdlib-2_13-bootstrapped`).
+    settings(commonBootstrappedSettings).
+    settings(
+      moduleName := "scala-library",
+      exportJars := true,
+      Compile / compile := {
+        val classDir = (Compile / classDirectory).value
+
+        // Copy contents of Scala 2.13 JAR into classDir
+        val s = streams.value
+        val report = updateClassifiers.value
+        val scalaLibraryJar =
+          // FIXME could not find it with report.select. Other options?
+          report.select(
+              configuration = configurationFilter("compile"),
+              module = (_: ModuleID).name == "scala-library",
+              artifact = artifactFilter(`type` = "src"))
+              .headOption.getOrElse {
+            sys.error(s"Could not fetch scala-library sources")
+          }.getParentFile / s"scala-library-${stdlibVersion(Bootstrapped)}.jar"
+        val cacheDir = s.cacheDirectory
+        FileFunction.cached(cacheDir / "fetchScalaLibraryBin",
+            FilesInfo.lastModified, FilesInfo.exists) { dependencies =>
+          s.log.info(s"Unpacking scala-library classfiles to $classDir")
+          if (classDir.exists)
+            IO.delete(classDir)
+          IO.createDirectory(classDir)
+          IO.unzip(scalaLibraryJar, classDir)
+          (classDir ** "*").get.toSet
+        } (Set(scalaLibraryJar)).toSeq
+        IO.delete(classDir / "META-INF")
+        // TODO What to do with library.properties, LICENSE, NOTICE, rootdoc.txt? Keep? Modify?
+
+        // Copy .tasty files from stdlib-2_13-bootstrapped into classDir
```

Comment:
The usual way to add files to the output jar is via `mappings`: https://www.scala-sbt.org/1.x/docs/Howto-Package.html#Modify+the+contents+of+the+package, this is likely to be more robust for interactive compilation etc.